src/community/label_propagation.jl

```diff
@@ -0,0 +1,99 @@
+"""
+Community detection using the label propagation algorithm (see [Raghavan et al.](http://arxiv.org/abs/0709.2938)).
+`g`: imput Graph
+`maxiter`: optional, maximum iterations
+return : array containing vertex assignments, the range of the output as 1:countdistinct(membership)
+"""
+function label_propagation(g::SimpleGraph; maxiter=1000)
+    n = nv(g)
+    membership = collect(1:n)
+    active_nodes = Set(vertices(g))
+    nc = NeighComm(collect(1:n), fill(-1,n), 1)
+    i = 0
+    while !isempty(active_nodes) && i < maxiter
+    	i += 1
+        random_order = shuffle(collect(active_nodes))
+        for u in random_order
+          old_comm = membership[u]
+          membership[u] = vote!(g, membership, nc, u)
+            if old_comm != membership[u]
+                for v in out_neighbors(g, u)
+                    push!(active_nodes, v)
+                end
+            else
+                delete!(active_nodes, u)
+            end
+        end
+    end
+    renumber_labels!(membership)
+    membership
+end
+
+"""Fast shuffle Array `a` in UnitRange `r` inplace."""
+function range_shuffle!(r::UnitRange, a::AbstractVector)
+	for i=length(r):-1:2
+		j = rand(1:i)
+		ii = i + r.start - 1
+		jj = j + r.start - 1
+		a[ii],a[jj] = a[jj],a[ii]
+	end
+end
```

Comment:
This doesn't address the correctness. 

```julia

function randperm(r::AbstractRNG, n::Integer)
    a = Array(typeof(n), n)
    @assert n <= Int64(2)^52
    if n == 0
       return a
    end
    a[1] = 1
    mask = 3
    @inbounds for i = 2:Int(n)
        j = 1 + rand_lt(r, i, mask)
        a[i] = a[j]
        a[j] = i
        i == 1+mask && (mask = 2mask + 1)
    end
    return a
end
```
Base uses this `mask` variable and `rand_lt` function. I assume it is there for good reason. Random number generation is harder than it looks so I want to be extra careful that it is correct. Also if you can check `start(r)>0 && end(r) < length(a)` at the beginning, then you can turn on `@inbounds` for the loop.